agent/engine/docker_task_engine.go

```diff
@@ -288,6 +323,96 @@ func (engine *DockerTaskEngine) Init(ctx context.Context) error {
 	return nil
 }
 
+// Always wakes up when at least one event arrives on buffered channel monitorQueuedTaskEvent
+// but does not block if monitorQueuedTasks is already processing queued tasks
```

Comment:
Can we elaborate a little more in the comment on 1) when we will be invoking this method (who will be sending messages onto the channel), and 2) why is buffer size of one sufficient (why's it okay to drop any additional messages)